### PR TITLE
Feature/vr 4033 experiments batch get in lineage and integration test with 2 users

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java [hunk 1]
```diff
@@ -51,7 +52,8 @@
 public class LineageServiceImpl extends LineageServiceImplBase {
 
   private static final Logger LOGGER = LogManager.getLogger(LineageServiceImpl.class);
-  private final ExperimentRunDAO experimentDAO;
+  private static final String EXPERIMENT_RUN_NOT_FOUND = "Experiment run not found";
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java]: Add messages on ModelDBConstant.java file

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java [hunk 2]
```diff
@@ -223,46 +225,66 @@ private Response filterInputOutput(Session session, Response response) {
...
+        List<LineageEntry> itemList = lineageEntryBatchResponseSingle.getItemsList();
+        for (LineageEntry lineageEntry : itemList) {
+          if (lineageEntry.getDescriptionCase() == DescriptionCase.EXPERIMENT_RUN) {
+            unfilteredExperimentRunIds.add(lineageEntry.getExperimentRun());
+          }
+        }
+      }
+    }
+    final Set<String> filteredExperimentRunIds;
+    if (unfilteredExperimentRunIds.size() == 0) {
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java]: As per intellij IDE suggestion, good to have 
```suggestion
    if (unfilteredExperimentRunIds.isEmpty()) {
```

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java [hunk 3]
```diff
@@ -299,33 +325,91 @@ private boolean filterLineageEntry(
...
   private void checkResourcesExistsAndAccessible(Session session, List<LineageEntry> lineageEntries)
       throws ModelDBException, NoSuchAlgorithmException, InvalidProtocolBufferException {
-    Set<String> experimentRuns = new HashSet<>();
+    Set<String> experimentRunIds = new HashSet<>();
     Map<Long, RepositoryContainer> blobs = new HashMap<>();
     for (LineageEntry lineageEntry : lineageEntries) {
-      validate(session, experimentRuns, blobs, lineageEntry);
+      validate(session, experimentRunIds, blobs, lineageEntry);
+    }
+    if (experimentRunIds.size() != 0) {
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java]:
```suggestion
    if (!experimentRunIds.isEmpty()) {
```

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java [hunk 4]
```diff
@@ -299,33 +325,91 @@ private boolean filterLineageEntry(
...
+        LOGGER.warn(EXPERIMENT_RUN_NOT_FOUND);
+      } else {
+        try {
+          roleService.validateEntityUserWithUserInfo(
+              ModelDBServiceResourceTypes.PROJECT,
+              experimentRunEntity.getProject_id(),
+              ModelDBServiceActions.READ);
+          result.add(experimentRunEntity.getId());
+        } catch (StatusRuntimeException e) {
+          LOGGER.warn("Can't access entity {}", e.getMessage());
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java:371]: Is this expected to skip throwing an exception?

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java [hunk 5]
```diff
@@ -299,33 +325,91 @@ private boolean filterLineageEntry(
...
-          ExperimentRunEntity experimentRunEntity =
-              experimentDAO.getExperimentRun(session, experimentRun);
-          experimentRuns.add(experimentRun);
-          // Validate if current user has access to the entity or not
-          roleService.validateEntityUserWithUserInfo(
-              ModelDBServiceResourceTypes.PROJECT,
-              experimentRunEntity.getProject_id(),
-              ModelDBServiceActions.READ);
+        experimentRunIds.add(lineageEntry.getExperimentRun());
+        if (filteredExperimentRunIds == null) {
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java:427]: if list is empty then what is expected? if it is empty and still it should be return true then if looks like below suggestion
```suggestion
        if (filteredExperimentRunIds == null || filteredExperimentRunIds.isEmpty()) {
```

File: backend/src/test/java/ai/verta/modeldb/LineageTest.java [hunk 6]
```diff
@@ -360,6 +381,9 @@ public void createAndDeleteLineageTest() throws ModelDBException, NoSuchAlgorith
 
     long repositoryId =
         RepositoryTest.createRepository(versioningServiceBlockingStub, RepositoryTest.NAME);
+    long repositoryId2 =
+        RepositoryTest.createRepository(
+            versioningServiceBlockingStubClient2, RepositoryTest.NAME + "2");
```
Comment (anandjakhaniya) [inline backend/src/test/java/ai/verta/modeldb/LineageTest.java:397]: I think the below test cases do not work well with OSS. please check it with OSS once.

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java (hunk 4 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java:371]: yes

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java (hunk 5 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java:427]: I think if empty it should return false

File: backend/src/test/java/ai/verta/modeldb/LineageTest.java (hunk 6 above)
Comment (evgeniilezhnin) [reply backend/src/test/java/ai/verta/modeldb/LineageTest.java:397]: It works with OSS. There are some cases that are disabled in OSS test run by
`          if (app.getAuthServerHost() != null && app.getAuthServerPort() != null) {
`
but besides that everything will work

File: backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java]: done

**Comment by jkwatson-verta** [comment]
closing as this is quite out of date

**Pull request closed without merging**